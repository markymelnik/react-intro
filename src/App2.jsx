import React, { useState } from 'react';

const App2 = () => {
  const [heading, setHeading] = useState('Green');

  const clickHandler = () => {
    setHeading('Blue');
  };

  return (
    <>
      <button type='button' onClick={clickHandler}>
        Click Me
      </button>
      <h1>{heading}</h1>
    </>
  );
};

export default App2;