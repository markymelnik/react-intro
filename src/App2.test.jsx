import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App2 from './App2';

describe('App component', () => {
  it('renders green', () => {
    const { container } = render(<App2 />);
    expect(container).toMatchSnapshot();
  });

  it('renders blue after button click', async () => {
    const user = userEvent.setup();

    render(<App2 />);
    const button = screen.getByRole('button', { name: 'Click Me' });

    await user.click(button);

    expect(screen.getByRole('heading', { name: 'Blue' }));
  });
});