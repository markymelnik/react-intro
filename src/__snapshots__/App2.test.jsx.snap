// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders green 1`] = `
<div>
  <button
    type="button"
  >
    Click Me
  </button>
  <h1>
    Green
  </h1>
</div>
`;
